engine/src/main/java/org/destinationsol/menu/NewGameScreen.java

```diff
@@ -58,7 +60,10 @@
 
     @Override
     public void onAdd(SolApplication solApplication) {
-        continueControl.setEnabled(SaveManager.hasPrevShip("prevShip.ini"));
+        IniReader reader = new IniReader(Const.SAVE_FILE_NAME, null);
+        String saveMajorVersion = reader.getString("version", "").split("\\.")[0];
+        String gameMajorVersion = Const.VERSION.split("\\.")[0];
+        continueControl.setEnabled(SaveManager.hasPrevShip("prevShip.ini") && saveMajorVersion.equals(gameMajorVersion));
```

Comment:
"prevShip.ini" shouldn't be used here, now that you have the constant defined. In fact, I don't think the file should be passed to the save manager, as that's more of a save manager concern. I would consider moving all of this logic into the hasPrevShip function.